From 4bde4c3968767f4916c8bd3892c0511f6cfd1415 Mon Sep 17 00:00:00 2001
From: Johannes Bornhold <johannes.bornhold.extern@univention.de>
Date: Thu, 27 Mar 2025 13:24:52 +0100
Subject: [PATCH 08/10] allow to set individual attributes to readonly in
 passwordreset

Removed a "sed" based patch and replaced with this change instead.
---
 .../umc/python/passwordreset/__init__.py                        | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/management/univention-self-service/umc/python/passwordreset/__init__.py b/management/univention-self-service/umc/python/passwordreset/__init__.py
index 241de8b0c3..6b25ca4e92 100644
--- a/management/univention-self-service/umc/python/passwordreset/__init__.py
+++ b/management/univention-self-service/umc/python/passwordreset/__init__.py
@@ -441,7 +441,7 @@ class Instance(Base):
                 'syntax': prop.syntax.name,
                 'size': prop.size or prop.syntax.size,
                 'required': bool(prop.required),
-                'editable': bool(prop.may_change),
+                'editable': bool(prop.may_change and not ucr.is_true(f'self-service/udm_attributes/{propname}/readonly')),
                 'readonly': not bool(prop.editable) or propname in read_only_attributes,
                 'multivalue': bool(prop.multivalue),
             }
-- 
2.47.3

